File: modules/services/app/dao/UserIdentity.scala
package dao

import securesocial.core.AuthenticationMethod
import securesocial.core.IdentityId
import securesocial.core.OAuth1Info
import securesocial.core.OAuth2Info
import securesocial.core.PasswordInfo


/**
 * This trait provides access to the User associated to a certain Identity
 */
trait UserIdentity extends securesocial.core.Identity {

  val identity : models.Identity

  def user     : models.User = identity.user


  //
  // implements securesocial.core.Identity
  //

  override def identityId: IdentityId = IdentityId(identity.username, identity.provider)

  override def firstName: String = identity.firstName

  override def lastName: String  = identity.lastName

  override def fullName: String  = identity.fullName

  override def email: Option[String]     = Some(identity.email)

  override def avatarUrl: Option[String] = if (identity.hasAvatarUrl) Option(identity.avatarUrl) else None

  override def authMethod: AuthenticationMethod = AuthenticationMethod(identity.authMethod)

  override def oAuth1Info: Option[OAuth1Info] =
    if (identity.hasOAuth1Info)
      Option(OAuth1Info(identity.oAuth1InfoToken, identity.oAuth1InfoSecret))
    else
      None

  override def oAuth2Info: Option[OAuth2Info] =
    if (identity.hasOAuth2Info)
      Option(OAuth2Info(
        identity.oAuth2InfoAccessToken,
        Option(identity.oAuth2InfoTokenType),
        if (identity.oAuth2InfoExpiresIn <= 0) None else Option(identity.oAuth2InfoExpiresIn),
        Option(identity.oAuth2InfoRefreshToken)))
    else
      None

  override def passwordInfo: Option[PasswordInfo] =
    if (identity.hasPasswordInfo)
      Option(PasswordInfo(
        identity.passwordInfoHasher,
        identity.passwordInfoPassword,
        Option(identity.passwordInfoSalt)))
    else
      None

}

File: modules/models/app/dao/UserIdentity.java
package dao;

import scala.None;
import scala.Option;
import scala.Option$;
import securesocial.core.AuthenticationMethod;
import securesocial.core.IdentityId;
import securesocial.core.OAuth1Info;
import securesocial.core.OAuth2Info;
import securesocial.core.PasswordInfo;


/**
 * This trait provides access to the User associated to a certain Identity
 */
public class UserIdentity implements securesocial.core.Identity {

    private final models.Identity identity;
    private final models.User     user;


    public UserIdentity(models.Identity identity) {
        this.identity = identity;
        this.user = identity.user;
    }

    public models.User user() {
        return this.user;
    }

    //
    // implements securesocial.core.Identity
    //

    @Override
    public IdentityId identityId() {
        return new IdentityId(identity.username, identity.provider);
    }

    @Override
    public String firstName() {
        return identity.firstName;
    }

    @Override
    public String lastName() {
        return identity.lastName;
    }

    @Override
    public String fullName() {
        return identity.fullName;
    }

    @Override
    public scala.Option<String> email() {
        return scala.Option$.MODULE$.apply(identity.email);
    }

    @Override
    public scala.Option<String> avatarUrl() {
        return (!identity.hasAvatarUrl) ? null: scala.Option$.MODULE$.apply(identity.avatarUrl);
    }

    @Override
    public AuthenticationMethod authMethod() {
        return new AuthenticationMethod(identity.authMethod);
    }

    @Override
    public Option<OAuth1Info> oAuth1Info() {
        return (identity.hasOAuth1Info)
                ? scala.Option$.MODULE$.apply(
                        new OAuth1Info(
                                identity.oAuth1InfoToken,
                                identity.oAuth1InfoSecret))
                : null;
    }

    @Override
    public Option<OAuth2Info> oAuth2Info() {
        return (identity.hasOAuth1Info)
                ? scala.Option$.MODULE$.apply(
                        new OAuth2Info(
                                identity.oAuth2InfoAccessToken,
                                scala.Option$.MODULE$.apply(identity.oAuth2InfoTokenType),
                                scala.Option$.MODULE$.apply((Object)identity.oAuth2InfoExpiresIn),
                                scala.Option$.MODULE$.apply(identity.oAuth2InfoRefreshToken)
                        )
                )
                : null;
    }

    @Override
    public Option<PasswordInfo> passwordInfo() {
        return (identity.hasPasswordInfo)
                ? scala.Option$.MODULE$.apply(
                        new PasswordInfo(
                                identity.passwordInfoHasher,
                                identity.passwordInfoPassword,
                                scala.Option$.MODULE$.apply(identity.passwordInfoSalt)))
                : null;
    }

}